### PR TITLE
Updates to prepare for integrating PeerManager with Splinterd/Admin Service

### PR DESCRIPTION
**Pull request by agunde406**

### CODE REVIEW

File: libsplinter/src/network/peer_manager/connector.rs [hunk 1]
```diff
@@ -97,6 +97,37 @@ impl PeerManagerConnector {
...
+    /// Request that a peer is added to the PeerManager. If a peer already exists, the peer's ref
+    /// count will be incremented. This function should be used when the peer id is unknown.
+    ///
+    /// # Arguments
+    ///
+    /// * `endpoint` -  The  endpoints associated with the peer.
+    ///
+    /// Returns a PeerRef, that when dropped, will automatically send a removal request to the
+    /// PeerManager.
+    pub fn add_unidentified_peer(&self, endpoint: String) -> Result<PeerRef, PeerRefAddError> {
```
Comment (peterschwarz) [inline libsplinter/src/network/peer_manager/connector.rs:109]: The commit message doesn't reflect this change.

Comment (agunde406) [reply libsplinter/src/network/peer_manager/connector.rs:109]: Whoops, fixed (old commit message)

File: libsplinter/src/network/peer_manager/connector.rs [hunk 2]
```diff
@@ -97,6 +97,37 @@ impl PeerManagerConnector {
             .map_err(|err| PeerRefAddError::ReceiveError(format!("{:?}", err)))?
     }
 
+    /// Request that a peer is added to the PeerManager. If a peer already exists, the peer's ref
+    /// count will be incremented. This function should be used when the peer id is unknown.
+    ///
+    /// # Arguments
+    ///
+    /// * `endpoint` -  The  endpoints associated with the peer.
```
Comment (ltseeley) [inline libsplinter/src/network/peer_manager/connector.rs:105]: Should this be "endpoint" or "endpoints"? The description of the argument says "endpoints" (also looks like a double space here)

File: libsplinter/src/network/connection_manager/authorizers.rs [hunk 3]
```diff
@@ -85,7 +85,7 @@ impl Authorizer for InprocAuthorizer {
 /// the empty string as a prefix, but should be added last.
 #[derive(Default)]
 pub struct Authorizers {
-    authorizers: Vec<(String, Box<dyn Authorizer>)>,
+    authorizers: Vec<(String, Box<dyn Authorizer + Send>)>,
```
Comment (ltseeley) [inline libsplinter/src/network/connection_manager/authorizers.rs:88]: Typo in commit message: s/Add to/Add/

File: libsplinter/src/circuit/service.rs [hunk 4]
```diff
@@ -70,7 +70,7 @@ impl Service {
 
 #[derive(Debug, PartialEq, Serialize, Deserialize, Clone)]
 pub struct SplinterNode {
-    #[serde(skip)]
+    #[serde(default)]
```
Comment (ltseeley) [inline libsplinter/src/circuit/service.rs:73]: Typos in commit message: s/no/now/ and s/backwords/backwards/

File: libsplinter/src/network/peer_manager/connector.rs (hunk 2 above)
Comment (agunde406) [reply libsplinter/src/network/peer_manager/connector.rs]: should be endpoint, fixed

File: libsplinter/src/network/connection_manager/authorizers.rs (hunk 3 above)
Comment (agunde406) [reply libsplinter/src/network/connection_manager/authorizers.rs:88]: fixed

File: libsplinter/src/circuit/service.rs (hunk 4 above)
Comment (agunde406) [reply libsplinter/src/circuit/service.rs:73]: fixed

**Pull request merged**